### PR TITLE
feat(notifications): adds double write for weekly reports

### PR DESCRIPTION
**Pull request by scefali**
Need to write to weekly report options to the new `NotificationSettingOption` table

### CODE REVIEW

File: src/sentry/api/endpoints/user_notification_fine_tuning.py [hunk 1]
```diff
@@ -135,6 +143,16 @@ def _handle_put_reports(user, data):
             elif not enabled:
                 value.add(org_id)
 
+            if double_write_enabled:
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/user_notification_fine_tuning.py:146]: would be good to add a comment here explaining what's going on here / when this should be removed.

**Review by JoshFerge — COMMENTED** [review]
is there a technical spec / overarching issue for these changes? curious why double writes are needed here!

**Comment by scefali** [comment]
@JoshFerge see here: https://www.notion.so/sentry/Notification-Settings-Technical-Spec-fa5408ab2d9740068e59a3cfc0b1f2e6

The double write is pretty standard with migrations, no? After we run the data migration, we need to enable double writes to ensure the setting stay in sync.

**Comment by JoshFerge** [comment]
> @JoshFerge see here: https://www.notion.so/sentry/Notification-Settings-Technical-Spec-fa5408ab2d9740068e59a3cfc0b1f2e6
> 
> The double write is pretty standard with migrations, no? After we run the data migration, we need to enable double writes to ensure the setting stay in sync.

yes totally, that makes sense. thanks for that context!

**Review by snigdhas — APPROVED** [review]

**Pull request merged**